extensions/websocket/01_websocket.js

```diff
@@ -218,54 +250,67 @@
       }
 
       if (
-        protocols.some((protocol) => !HTTP_TOKEN_CODE_POINT_RE.test(protocol))
+        ArrayPrototypeSome(
+          protocols,
+          (protocol) =>
+            !RegExpPrototypeTest(HTTP_TOKEN_CODE_POINT_RE, protocol),
+        )
       ) {
         throw new DOMException(
           "Invalid protocol value.",
           "SyntaxError",
         );
       }
 
-      core.opAsync("op_ws_create", {
-        url: wsURL.href,
-        protocols: protocols.join(", "),
-      }).then((create) => {
-        this[_rid] = create.rid;
-        this[_extensions] = create.extensions;
-        this[_protocol] = create.protocol;
-
-        if (this[_readyState] === CLOSING) {
-          core.opAsync("op_ws_close", {
-            rid: this[_rid],
-          }).then(() => {
-            this[_readyState] = CLOSED;
-
-            const errEvent = new ErrorEvent("error");
-            this.dispatchEvent(errEvent);
-
-            const event = new CloseEvent("close");
-            this.dispatchEvent(event);
-            tryClose(this[_rid]);
-          });
-        } else {
-          this[_readyState] = OPEN;
-          const event = new Event("open");
-          this.dispatchEvent(event);
+      PromisePrototypeCatch(
```

Comment:
You can combine the "catch" into the `PromisePrototypeThen` by passing the catch handler as a second callback.

Reply:
done